### ISSUE TITLE
`FileConnector.partitionBy()` cannot handle duplicate columns 

### ISSUE THREAD

**Issue by qxzzxq** [JCDecaux/setl#122]
## Description

For a DataFrame like this:
```
+----------+----------+-----------+-----+
|partition1|partition2|clustering1|value|
+----------+----------+-----------+-----+
|         1|        p1|         c1|    1|
|         2|        p2|         c2|    2|
|         2|        p1|         c2|    2|
|         3|        p3|         c3|    3|
|         3|        p2|         c3|    3|
|         3|        p3|         c3|    3|
+----------+----------+-----------+-----+
```

If save the dataframe with the following connector and partition:

```scala
val csvConnector = new CSVConnector(
  Map[String, String](
    "path" -> "src/test/resources/test_csv",
    "inferSchema" -> "true",
    "header" -> "true",
    "saveMode" -> "Overwrite")
)

csvConnector
  .partitionBy("partition1", "partition1")
  .write(df)
```

We will have the following output:
![image](https://user-images.githubusercontent.com/17144939/84372044-d8fe8300-abda-11ea-95f8-8b6676b44aa3.png)

The subdirectory `partition1=X` is created twice in the output directory. This is the expected behaviour of Spark. But I wonder if would be useful to deduplicate the partition columns. Because the duplicated partition will cause schema inference problems for tools like AWS Glue crawler.

## Solution proposal

The idea it to be able to drop automatically duplicated partition columns, with the following constraints:
- keep the order of partition columns
- be able to select between dropping and keeping the duplicated partition columns (in order to keep the default behaviour of Spark)

@JorisTruong @maroil advise is needed

**Comment by qxzzxq** [comment]
FYI I created un ticket for spark https://issues.apache.org/jira/browse/SPARK-31968. Maybe this issue will be solved in spark. 

According to the new PR, an `AnalysisException` will be thrown if duplicate partitioning columns are detected.

**Comment by qxzzxq** [comment]
This problem will be fixed in Spark 2.4.7 and 3.0+
